connexta/geospatialdraw
# Fix point radius shift-click capture and remove compiled code from repo

#### What does this PR do, and why?
When holding shift and creating a point radius, an error `to point is required` can be seen in the console. The result is that onUpdate isn't triggered, which can break some downstream workflows. 

This occurs because the default [ol.interaction.Draw](https://geoadmin.github.io/ol3/apidoc/ol.interaction.Draw.html)'s `freehandCondition` captures the shift click and attempts to enable free draw. This appears to interrupt the normal point radius workflow, particularly onUpdate, and results in  the interal linestring (used to create the point radius) having only the `from` coordinate, but not the `to` coordinate.

This pr prevents the freehandCondition, and condition, from capturing these clicks and breaking workflow. If these buttons are desired, we'll want to think through how we can allow them without breaking the workflow (perhaps another option on the drawing menu)

Note: this pr also removes the compiled code from the repo. We'll save that for tags and releases.

#### Testing (using storybook)

1. Pull this pr and run `yarn storybook`
2. In storybook, navigate to the "full featured" demo and change the shape to point radius
    ![image](https://user-images.githubusercontent.com/22667297/117732935-92782680-b1a5-11eb-94fb-6f698771fc5d.png)
3. Open dev tools to the console
4. Hold shift and draw the point radius.
5. Verify that no errors appear in the console.

#### Testing (downstream)

1. Pull this pr and run `yarn build:all`.  Note: you may have to `yarn install` first, if you yarn doesn't recognize `rimraf`
2. In your downstream project, point your local geospatial draw in package.json (e.g. `geospatialdraw: "C:\\projects\\geospatialdraw"`)
3. In your downstream project, yarn install to grab the local pr.
4. Run your downstream project, navigate to your point radius map feature, and verify that drawing a point radius works as expected when you hold shift.
